### PR TITLE
Linspace, forEach: make sure they can be constexpr-evaluated

### PR DESCRIPTION
**Pull request by pmglr**

### CODE REVIEW

**Review by markus-sonible — APPROVED** [review]
looks fine to me!

File: include/vctr/Containers/VctrBase.h [hunk 1]
```diff
@@ -20,6 +20,11 @@
   ==============================================================================
 */
 
+#if VCTR_MSVC
+__pragma(warning(push))
+    __pragma(warning(disable:4324))
```
Comment (JanosGit) [inline include/vctr/Containers/VctrBase.h]: Funky indentation..?

File: include/vctr/Containers/VctrBase.h [hunk 2]
```diff
@@ -303,7 +308,7 @@ class VctrBase : private Config,
 
         const auto increment = (stop - start) / double (num - size_t (includeEnd));
 
-        [[maybe_unused]] const auto isIntegerIncrement = increment - std::floor (increment) == ElementType (0);
+        [[maybe_unused]] const auto isIntegerIncrement = increment - gcem::floor (increment) == ElementType (0);
```
Comment (JanosGit) [inline include/vctr/Containers/VctrBase.h:315]: GCEM is only an optional dependency, so every code using it should provide a fallback solution for cases when GCEM is not available

**Review by JanosGit — CHANGES_REQUESTED** [review]
Good additions, but please make sure that this also works without GCEM

File: include/vctr/Containers/VctrBase.h (hunk 1 above)
Comment (pmglr) [reply include/vctr/Containers/VctrBase.h]: Oops

Comment (pmglr) [reply include/vctr/Containers/VctrBase.h]: The indentation is actually clang-format conform. However not the missing spaces

Comment (JanosGit) [reply include/vctr/Containers/VctrBase.h]: I'd rather wrap that in clang-format guards and align it manually, don't see any good reason why this should look like this, I rather consider this a clang-format fault

**Review by JanosGit — APPROVED** [review]

**Pull request merged**